Work around missing musl and gnu platform configs in LLVM's config.bzl.

LLVM's Bazel overlay references @llvm//platforms/config:musl and
@llvm//platforms/config:gnu which fail resolution when building without
custom LLVM platform configurations. We replace them with standard
@platforms//os:linux.

Precedents in other OSS projects:
- https://github.com/google/xls/blob/64bf93a3963dc6fd55b749e457e3345c91a5091f/dependency_support/llvm/platforms_config.patch
- https://github.com/carbon-language/carbon-lang/blob/f58de06c0dbd506573620d8f2873923a8697c925/bazel/llvm_project/0011_Temporarily_remove_reference_to_hermetic_toolchain.patch

diff --git a/utils/bazel/llvm-project-overlay/llvm/config.bzl b/utils/bazel/llvm-project-overlay/llvm/config.bzl
--- a/utils/bazel/llvm-project-overlay/llvm/config.bzl
+++ b/utils/bazel/llvm-project-overlay/llvm/config.bzl
@@ -49,13 +49,12 @@
 backtrace_defines = select({
     "@platforms//os:windows": [],
-    "@llvm//platforms/config:musl": [],
     "//conditions:default": [
         "HAVE_BACKTRACE=1",
         "BACKTRACE_HEADER=<execinfo.h>",
     ],
 })
 
 mallinfo_defines = select({
-    "@llvm//platforms/config:gnu": ["HAVE_MALLINFO=1"],
+    "@platforms//os:linux": ["HAVE_MALLINFO=1"],
     "//conditions:default": [],
 })
